### PR TITLE
Add an entry function to update the federated jwkset

### PR DESCRIPTION
**Pull request by heliuchuan**
## Description

Add an entry function to the framework that allows for updating an issuers JWK set.

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [x] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/jwks.move [hunk 1]
```diff
@@ -201,6 +201,28 @@ module aptos_framework::jwks {
         assert!(num_bytes < MAX_FEDERATED_JWKS_SIZE_BYTES, error::invalid_argument(EFEDERATED_JWKS_TOO_LARGE));
     }
 
+    // Todo: description
+    public fun update_federated_jwk_set(jwk_owner: &signer, iss: vector<u8>, kid_vec: vector<String>, alg_vec: vector<String>, e_vec: vector<String>, n_vec: vector<String>) acquires FederatedJWKs {
```
Comment (alinush) [inline aptos-move/framework/aptos-framework/sources/jwks.move:205]: Is this necessary? Why are you adding it?

Comment (alinush) [inline aptos-move/framework/aptos-framework/sources/jwks.move:205]: You want this to be `entry`, right?

Comment (heliuchuan) [reply aptos-move/framework/aptos-framework/sources/jwks.move:205]: Discussed offline.  Can be done via script but given this is a common usecase and scripts require move compilation it is good to have it in the framework

Comment (heliuchuan) [reply aptos-move/framework/aptos-framework/sources/jwks.move:205]: yup

File: aptos-move/framework/aptos-framework/sources/jwks.move [hunk 2]
```diff
@@ -201,6 +202,29 @@ module aptos_framework::jwks {
         assert!(num_bytes < MAX_FEDERATED_JWKS_SIZE_BYTES, error::invalid_argument(EFEDERATED_JWKS_TOO_LARGE));
     }
 
+    // This can be called to install or update a set of JWKs for a federated OIDC provider.  This function should
+    // be invoked to intially install a set of JWKs or to update a set of JWKs when a keypair is rotated.
```
Comment (gregnazario) [inline aptos-move/framework/aptos-framework/sources/jwks.move:206]: These should be doc comments `///`

File: aptos-move/framework/aptos-framework/sources/jwks.move [hunk 3]
```diff
@@ -201,6 +202,29 @@ module aptos_framework::jwks {
...
+    public entry fun update_federated_jwk_set(jwk_owner: &signer, iss: vector<u8>, kid_vec: vector<String>, alg_vec: vector<String>, e_vec: vector<String>, n_vec: vector<String>) acquires FederatedJWKs {
```
Comment (gregnazario) [inline aptos-move/framework/aptos-framework/sources/jwks.move:207]: Can we use descriptive names for these inputs?  And describe them in the documentation?

What are the requirements for the inputs?  How do they relate to each other?

This requires a lot of context to understand all of them.

Comment (gregnazario) [inline aptos-move/framework/aptos-framework/sources/jwks.move:207]: Which AIP does this apply to?

Comment (alinush) [reply aptos-move/framework/aptos-framework/sources/jwks.move:207]: [AIP-96](https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-96.md).

@heliuchuan, add to comments?

**Review by alinush — APPROVED** [review]
...but address @gregnazario's comments!

File: aptos-move/framework/aptos-framework/sources/jwks.move (hunk 3 above)
Comment (heliuchuan) [reply aptos-move/framework/aptos-framework/sources/jwks.move:256]: Updated.  Not sure what you mean by descriptive names.  I at least added thorough description.

File: aptos-move/framework/aptos-framework/sources/jwks.move [hunk 4]
```diff
@@ -201,6 +202,78 @@ module aptos_framework::jwks {
...
+    ///     {
+    ///       "kty": "RSA",
+    ///       "kid": "b2620d5e7f132b52afe8875cdf3776c064249d04",
+    ///       "alg": "RS256",
+    ///       "n": "pi22xDdK2fz5gclIbDIGghLDYiRO56eW2GUcboeVlhbAuhuT5mlEYIevkxdPOg5n6qICePZiQSxkwcYMIZyLkZhSJ2d2M6Szx2gDtnAmee6o_tWdroKu0DjqwG8pZU693oLaIjLku3IK20lTs6-2TeH-pUYMjEqiFMhn-hb7wnvH_FuPTjgz9i0rEdw_Hf3Wk6CMypaUHi31y6twrMWq1jEbdQNl50EwH-RQmQ9bs3Wm9V9t-2-_Jzg3AT0Ny4zEDU7WXgN2DevM8_FVje4IgztNy29XUkeUctHsr-431_Iu23JIy6U4Kxn36X3RlVUKEkOMpkDD3kd81JPW4Ger_w",
+    ///       "e": "AQAB",
+    ///       "use": "sig"
+    ///     }
+    ///   ]
+    /// }
```
Comment (gregnazario) [inline aptos-move/framework/aptos-framework/sources/jwks.move:232]:
````suggestion
    /// ```
    /// {
    ///   "keys": [
    ///     {
    ///       "alg": "RS256",
    ///       "use": "sig",
    ///       "kty": "RSA",
    ///       "n": "wNHgGSG5B5xOEQNFPW2p_6ZxZbfPoAU5VceBUuNwQWLop0ohW0vpoZLU1tAsq_S9s5iwy27rJw4EZAOGBR9oTRq1Y6Li5pDVJfmzyRNtmWCWndR-bPqhs_dkJU7MbGwcvfLsN9FSHESFrS9sfGtUX-lZfLoGux23TKdYV9EE-H-NDASxrVFUk2GWc3rL6UEMWrMnOqV9-tghybDU3fcRdNTDuXUr9qDYmhmNegYjYu4REGjqeSyIG1tuQxYpOBH-tohtcfGY-oRTS09kgsSS9Q5BRM4qqCkGP28WhlSf4ui0-norS0gKMMI1P_ZAGEsLn9p2TlYMpewvIuhjJs1thw",
    ///       "kid": "d7b939771a7800c413f90051012d975981916d71",
    ///       "e": "AQAB"
    ///     },
    ///     {
    ///       "kty": "RSA",
    ///       "kid": "b2620d5e7f132b52afe8875cdf3776c064249d04",
    ///       "alg": "RS256",
    ///       "n": "pi22xDdK2fz5gclIbDIGghLDYiRO56eW2GUcboeVlhbAuhuT5mlEYIevkxdPOg5n6qICePZiQSxkwcYMIZyLkZhSJ2d2M6Szx2gDtnAmee6o_tWdroKu0DjqwG8pZU693oLaIjLku3IK20lTs6-2TeH-pUYMjEqiFMhn-hb7wnvH_FuPTjgz9i0rEdw_Hf3Wk6CMypaUHi31y6twrMWq1jEbdQNl50EwH-RQmQ9bs3Wm9V9t-2-_Jzg3AT0Ny4zEDU7WXgN2DevM8_FVje4IgztNy29XUkeUctHsr-431_Iu23JIy6U4Kxn36X3RlVUKEkOMpkDD3kd81JPW4Ger_w",
    ///       "e": "AQAB",
    ///       "use": "sig"
    ///     }
    ///   ]
    /// }
    /// ```
````

File: aptos-move/framework/aptos-framework/sources/jwks.move [hunk 5]
```diff
@@ -201,6 +202,78 @@ module aptos_framework::jwks {
         assert!(num_bytes < MAX_FEDERATED_JWKS_SIZE_BYTES, error::invalid_argument(EFEDERATED_JWKS_TOO_LARGE));
     }
 
+    /// This can be called to install or update a set of JWKs for a federated OIDC provider.  This function should
+    /// be invoked to intially install a set of JWKs or to update a set of JWKs when a keypair is rotated.
+    ///
+    /// The `iss` parameter is the value of the `iss` claim on the JWTs that are to be verified by the JWK set.
+    /// `kid_vec`, `alg_vec`, `e_vec`, `n_vec` are String vectors of the JWK attributes `kid`, `alg`, `e` and `n` respectively.
+    ///
```
Comment (gregnazario) [inline aptos-move/framework/aptos-framework/sources/jwks.move:210]: I'm guessing this is all in the original documentation for `patch_federated_jwks`?  Though I don't see it.

What is `iss`?

* `alg` is the `algorithm`
* `kty` is the `key type`
* `kid` is `key id`?
* `e` is some cryptographic `e`
* `n` is some cryptographic `n`

**Review by gregnazario — COMMENTED** [review]
We actually are using the `Any` type?

I thought we had avoided that.  Why not use an enum (or something that will look like an enum)?

I see the usage of this function, to build up all the internal types, just let's please ensure that we have documentation around all the different short names on the structs accordingly.

**Review by gregnazario — APPROVED** [review]

**Pull request merged**